esx_service/vmdk_ops.py

```diff
@@ -402,8 +402,15 @@ def vol_info(vol_meta, vol_size_info, datastore):
 # Return error, or None for OK
 def removeVMDK(vmdk_path):
     logging.info("*** removeVMDK: %s", vmdk_path)
-    cmd = "{0} {1}".format(VMDK_DELETE_CMD, vmdk_path)
 
+    # Check the current volume status
+    attached, uuid, attach_as = getStatusAttached(vmdk_path)
+    if attached:
```

Comment:
If a volume is attached and VM is powered off then lets do this,

a. check if the volume is attached and
b. check if the VM is powered off and if yes then detach the disk from the VM.

And all of this happens already - vmdk_ops.py:handle_stale_attach() :-)